service/authorization.js

```diff
@@ -0,0 +1,44 @@
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
+const { getUserByEmail, addUserToken, getUserById } = require("./functions");
+const secret = process.env.JWT_SECRET; 
+
+const createToken = (user) => {
+  const payload = {
+    id: user._id,
+  };
+  const token = jwt.sign(payload, secret);
+  return token;
+};
+  
+const handleLogin = async (email, password) => {
+  const user = await getUserByEmail(email);
+  const userPassword = user.password;
+  const result = bcrypt.compareSync(password, userPassword);
+  if (result) {
+    const token = createToken(user);
+    await addUserToken(user._id, token);
+    return token;
+  }  
+};
+
+const auth = async (req, res, next) => {
```

Comment:
Ogólnie to działa i nawet dobrze, ale chcemy zastosować tutaj strategię oraz użyć jedngo z bardziej znanych modeli tej strategii czyli bearer tokenu (Tak w skrócie jest to prawie to samo tylko że z przodu jest słówko Bearer)

Pokazane jak to zrobić masz na lms w materiałach Moduł 4 lekcja 7-8 autoryzacja jwt (tam praktycznie wystarczy to skopiować, jedyne co mususz dodać to sprawdzanie czy token w req będzie taki sam jak w w db bo tego przykład nie obejmuje)